### DISCUSSION TITLE
Protocol not supported

### DISCUSSION THREAD

**Discussion by Dimonlipko** [cedricp/ddt4all#1312] [Q&A] [answered]
Hello) 

I have Megan 5 e-tech electric, and i need to clear dtc in EVC. When i try to open EVC ECU in ddt4all have error that protocol not supported. In console i have message "Cannot retrieve functionnal address of ECU N_PZ1A_HEVC_FD_PT_20210427 @ 18DADAF1"

![image_2024-12-08_23-41-09](https://github.com/user-attachments/assets/5bb1e7aa-4490-4678-831e-bc9e5de39a61)

**Comment by Furtif** [comment]
This is not à issue is gateway

**Reply by Dimonlipko** [reply]
I am try in demo mode, and ddt is don't open ECU. Gateway in car I bypassed, and connected directly to EVC.

**Reply by Furtif** [reply]
put # in line https://github.com/cedricp/ddt4all/blob/master/ecu.py#L1013 and try.

**Reply by Furtif** [reply]
but is bad, 18DADAF1 not is implemented in json is 29bits stuffs

**Reply by Furtif** [reply]
go make a dump for get 29bits

**Reply by Furtif** [reply]
[projects.json](https://github.com/user-attachments/files/18085201/projects.json)
test but not thinks is good for connected mode

**Reply by Furtif** [reply]
```
<Function Address="218" Name="(H) EVC-HCM-VCM_29b">
    <Name xml:lang="en">Vehicle Control Module_29bits</Name>
    <Name xml:lang="fr">Vehicle Control Module_29bits</Name>
    <XId>-1730487567</XId>
    <RId>-1730481702</RId>
    <ProtocolList>
      <Protocol Code="8" InitMode="3" MultiPoint="1" Name="DiagOnCANFD">
        <Address Present="0" Extended="0" Xid="7E4" Rid="7EC" />
        <Address Present="1" Extended="1" Xid="18DADAF1" Rid="18DAF1DA" />
        <LogicalAddress>10DA</LogicalAddress>
        <Search />
      </Protocol>
      <Protocol Code="6" InitMode="3" MultiPoint="1" Name="DiagOnCAN">
        <Address Present="0" Extended="0" Xid="7E4" Rid="7EC" />
        <Address Present="1" Extended="1" Xid="18DADAF1" Rid="18DAF1DA" />
        <LogicalAddress>10DA</LogicalAddress>
        <Search />
      </Protocol>
    </ProtocolList>
  </Function>
  ```
  normal
  `<Address Present="0" Extended="0" Xid="7E4" Rid="7EC" />`
  other (extended or 29bits, etc)
  `<Address Present="1" Extended="1" Xid="18DADAF1" Rid="18DAF1DA" />`

**Reply by Furtif** [reply, marked as answer]
[projects.json](https://github.com/user-attachments/files/18085400/projects.json)


try this is better and i thinks ok

**Reply by Furtif** [reply]
or try branch https://github.com/cedricp/ddt4all/tree/try_full_address

**Reply by Dimonlipko** [reply]
> [projects.json](https://github.com/user-attachments/files/18085201/projects.json) test but not thinks is good for connected mode

I copy this .json to my ddt4all, and now ECU is open) 

Tomorrow I will try with car.

Thanks!

**Reply by Furtif** [reply]
> > [projects.json](https://github.com/user-attachments/files/18085201/projects.json) test but not thinks is good for connected mode
> 
> I copy this .json to my ddt4all, and now ECU is open)
> 
> Tomorrow I will try with car.
> 
> Thanks!

i not think ok fallow pr opened https://github.com/cedricp/ddt4all/pull/1315

**Reply by Dimonlipko** [reply]
I check on the bench, elm327 + CAN Analyzer, and looks like it work. 

![image](https://github.com/user-attachments/assets/20edda95-f59a-403d-a9f9-991cc0c40361)

**Reply by Furtif** [reply]
is bad try code of branch my last commits

**Reply by Furtif** [reply]
https://github.com/cedricp/ddt4all/tree/try_full_address

**Reply by Furtif** [reply]
released https://github.com/cedricp/ddt4all/releases/tag/v3.0.3
